content/doc/book/pipeline/index.adoc

```diff
@@ -120,6 +143,76 @@ modeled in Jenkins Pipeline:
+This is the foundation of "Pipeline-as-Code"; treating the continuous delivery
+pipeline a part of the application to be version and reviewed like any other code.
+Creating a `Jenkinsfile` provides a number of immediate benefits:
+
+* Automatically create Pipelines for all Branches and Pull Requests
+* Code review/iteration on the Pipeline
+* Audit trail for the Pipeline
+* Single source of truth
+  footnote:[https://en.wikipedia.org/wiki/Single_Source_of_Truth]
+  for the Pipeline, which can be viewed and edited by multiple members of the project.
+
+
+While the syntax for defining a Pipeline, either in the web UI or with a
+`Jenkinsfile`, is the same, it's generally considered best practice to define
+the Pipeline in a `Jenkinsfile` and check that in to source control.
+
+[[syntax]]
+== Scripted Pipeline Syntax and Declarative Pipeline Syntax
+
+When Pipeline was first conceived it was natural to begin with Groovy as the
+foundation. Jenkins already had an embedded Groovy scripting engine with a console
+to interact with all parts of Jenkins and Groovy is a great language for creating
+a custom Domain-Specific Language (DSL). The
+link:https://plugins.jenkins.io/job-dsl[Job-DSL],
+plugin for Jenkins was also written in Groovy. This plugin lets you automate the
+creation and editing of any job in Jenkins, including Pipelines.
+
+Pipeline initially introduced key pipeline-specific concepts such as `node`,
+`stage`, `parallel`, and extension points to allow plugins to add other steps
+to the DSL but, otherwise didn't restrict the use of Groovy. This original syntax
+for creating Pipelines is now referred to as "Scripted Pipeline" and includes
+full programmatic control to allow scripting continuous delivery pipelines in Groovy.
+This givesPipeline creators tremendous flexibility in defining a pipeline and allows it to
+be extended via <<shared-libraries#, Shared Libraries>> or plugins.
+
+
+CAUTION: Due to the need to serialize all variables for durability some Groovy
+idioms are not fully supported yet. See
+https://issues.jenkins-ci.org/browse/JENKINS-27421[JENKINS-27421]
+and
+https://issues.jenkins-ci.org/browse/JENKINS-26481[JENKINS-26481]
+for more information.
+
+Writing Pipelines with Scripted Pipeline syntax, however, does require at least
+some proficiency with Groovy. Requiring all team members that touch the application's
+Pipeline to understand Groovy limits the ability for full-participation in
+code-review, audits, and editing of the Pipeline as part of the application code.
+Enter "Declarative Pipeline" syntax.
+
+Declarative Pipeline syntax was created to extend Pipeline to users of all experience
+levels and complement Scripted Pipeline syntax. As the name implies, it is intended
+to enable declarative programming
+footnoteref:[declarative, link:https://en.wikipedia.org/wiki/Declarative_programming[Declarative Programming]]
+for defining Pipelines as opposed to the imperative programming
```

Comment:
missing a comma (not required but would improve readability) "...for defining Pipelines, as opposed to the imperative..."